znai-server/src/main/java/com/twosigma/znai/server/preview/PreviewWebSocketHandler.java

```diff
@@ -67,6 +68,14 @@ public void sendMeta(DocMeta docMeta) {
         send(payload);
     }
 
+    public void sendDocReferences(DocReferences docReferences) {
```

Comment:
Could you explain briefly how websocket is working between backend and ui?

Reply:
websocket establishes on page load when the mode is `preview`. 
once we detect a file change we sent a json payload to a web client using `type`, and `payload` pattern. `payload` key changes based on `type`. 
`Documentation` react component has handlers for each type to handle.